src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```
@@ -356,11 +356,18 @@ public BitbucketCommit call() throws Exception {
     }
 
     private void setupClosureForPRBranch(BitbucketServerPullRequest pr) {
-        BitbucketServerBranch branch = (BitbucketServerBranch) pr.getSource().getBranch();
-        branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
-
-        branch = (BitbucketServerBranch) pr.getDestination().getBranch();
-        branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+        try {
+            BitbucketServerBranch branch = (BitbucketServerBranch) pr.getSource().getBranch();
+            if (branch != null) {
+                branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+            }
+            branch = (BitbucketServerBranch) pr.getDestination().getBranch();
+            if (branch != null) {
+                branch.setCommitClosure(new CommitClosure(branch.getRawNode()));
+            }
+        } catch (Exception e) {
```

Do you really want to catch ALL exceptions here?

---

Yup, I missed that 😱

---

@2improveIT ping

---

adapted the code